absl/absl.h

```diff
@@ -5,10 +5,41 @@
 
 namespace nda {
 
-// interval -> string.
+// interval -> string as the half-open interval [begin, end).
+//
+// Stringifies only the values, not whether they are static or dynamic.
 template <typename Sink, index_t Min_ = dynamic, index_t Extent_ = dynamic>
-void AbslStringify(Sink& sink, const nda::interval<Min_, Extent_>& p) {
-  absl::Format(&sink, "[%v, %v)", p.min(), p.min() + p.extent());
+void AbslStringify(Sink& sink, const interval<Min_, Extent_>& i) {
+  absl::Format(&sink, "[%v, %v)", i.min(), i.min() + i.extent());
+}
+
+// dim -> string as "dim(min, extent, stride)".
+//
+// Stringifies only the values, not whether they are static or dynamic.
+template <typename Sink, index_t Min_ = dynamic, index_t Extent_ = dynamic,
```

Comment:
I don't think you need the default template arguments here. I'm actually surprised it's not a compiler error. (Same above, missed it last time). If you don't need them, I suggest removing them purely for verbosity. I don't think the default arguments are ever helpful (these should be inferred if the argument is a `dim`, and if not, it won't matter).